### PR TITLE
Ccp4cloud

### PR DESCRIPTION
**Pull request by hlasimpk**
- ccp4 cloud specific changes, --ccp4cloud flag added that stops results page popping up and deletes unwanted files and stops local file locations being shown in the output results page. 
- Added in B-factor conversion for pLDDT scores and a sequence identity header used in MR with phaser.
- Added in coloured pfam domains where the colour corresponds to the pLDDT confidence score.

### CODE REVIEW

File: mrparse/mr_alphafold.py [hunk 1]
```diff
@@ -208,6 +211,53 @@ def get_plddt(struct):
...
+
+    for i, plddt in residues:
+        if plddt < 50:
+            v_low.append(i)
+        elif 70 > plddt > 50:
+            low.append(i)
+        elif 90 > plddt > 70:
+            confident.append(i)
+        elif plddt > 90:
+            v_high.append(i)
```
Comment (FilomenoSanchez) [inline mrparse/mr_alphafold.py:232]: You omitted the edge cases 50, 70 and 90. If `plddt` takes any of these values, then nothing happens.

File: mrparse/mr_alphafold.py [hunk 2]
```diff
@@ -208,6 +211,53 @@ def get_plddt(struct):
...
+
+    regions['v_low'] = _get_regions(v_low)
+    regions['low'] = _get_regions(low)
+    regions['confident'] = _get_regions(confident)
+    regions['v_high'] = _get_regions(v_high)
+
+    return regions
+
+
+def _get_regions(residues):
```
[the remark this answers is not in the document]
Comment (FilomenoSanchez) [reply mrparse/mr_alphafold.py:242]: Also I don't know if my alternative works in `ccp4-python v2.7`, I suspect not

File: mrparse/mr_analyse.py [hunk 3]
```diff
@@ -71,17 +72,20 @@ def run(seqin, hklin=None, run_serial=False, do_classify=True, pdb_dir=None, db_
...
+        shutil.rmtree(pdb_dir)
+    else:
+        opencmd = None
+        if sys.platform.lower().startswith('linux'):
+            opencmd = 'xdg-open'
+        elif sys.platform.lower().startswith('darwin'):
+            opencmd = 'open'
+        if opencmd:
+            subprocess.Popen([opencmd, html_out])
     return 0
```
Comment (FilomenoSanchez) [inline mrparse/mr_analyse.py:89]: I know is not code that changed in this PR, and I might just not be aware of the context of this function, but what's the point of always returning 0?

File: mrparse/mr_pfam.py [hunk 4]
```diff
@@ -52,24 +52,50 @@ def add_pfam_dict_to_models(models, sequence_length):
...
              }
+        motifs = []
+        colors = {'v_low': "ff7d45",
+                  "low": "ffdb13",
+                  "confident": "65cbf3",
+                  "v_high": "0053d6"}
+        for quality in plddt_regions.keys():
+            for plddt_region in plddt_regions[quality]:
+                plddt_region_start = plddt_region[0]
+                plddt_region_end = plddt_region[1]
```
Comment (FilomenoSanchez) [inline mrparse/mr_pfam.py:81]: Can be simplified as `plddt_region_start, plddt_region_end = plddt_region`

**Review by FilomenoSanchez — CHANGES_REQUESTED** [review]
A few minor comments about the code, I would say all of them are rather optional suggestions that you can safely ignore, except the one at `mrparse/mr_alphafold.py` at the function `def get_plddt_regions(struct, seqid_range)`. I think that could cause bugs. Also, regarding the implementation of the `--ccp4cloud` switch as a command line option, I'm not sure if this is the ideal solution. I understand why you use it, but since it is a very specific case, I doubt actual users will be interested in running with this option and it may just add noise to the command line arguments. I'm wondering, is there a specific environment variable set on CCP4 cloud that you could try to check to assert if the code is running on the cloud? That would be a cleaner solution but I understand adding the command line switch is the only option if there's no such variable in the CCP4 cloud environment.

File: mrparse/mr_alphafold.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hlasimpk) [reply mrparse/mr_alphafold.py:242]: This seems to work in v2.7, I'll swap it out

File: mrparse/mr_analyse.py (hunk 3 above)
Comment (hlasimpk) [reply mrparse/mr_analyse.py:89]: Not sure, this was written by Jens. 0 typically would signify a run code so if 0 is returned it probably indicates the code has executed successfully? It's probably not essential

**Review by FilomenoSanchez — APPROVED** [review]

**Pull request merged**